src/Paging.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react';
+
+export default class Paging extends Component {
+    constructor() {
+      super();
+      this.state = { page: 1 };
+    }
+  
+    componentDidMount() {
+      this.updateControls();
+  
+      window.addEventListener("hashchange", () => {
+        this.updateControls();
+      });
+    }
+  
+    updatePage(increment) {
+      const queryString = window.location.hash.slice(1);
+      const searchParams = new URLSearchParams(queryString);
+      searchParams.set("page", this.state.page + increment);
+  
+      window.location.hash = searchParams.toString();
+    }
+  
+    updateControls() {
+      const queryString = window.location.hash.slice(1);
+      const searchParams = new URLSearchParams(queryString);
+      let pageToUse = this.state.page;
+  
+      const parsedPage = Number(searchParams.get("page"));
+      if (isNaN(parsedPage)) {
+        pageToUse = 1;
+      } else {
+        pageToUse = parsedPage;
+      }
+  
+      this.setState({ page: pageToUse });
+    }
+  
+    render() {
+      const { page } = this.state;
+  
+      const perPage = 50; // this API can display more results per page
+      const { totalResults } = this.props;
+      const queryString = window.location.hash.slice(1);
+      const searchParams = new URLSearchParams(queryString);
+  
+      const parsedPage = parseInt(searchParams.get("page"));
+  
+      let pageToUse;
+      if (isNaN(parsedPage)) {
+        pageToUse = 1;
+      } else {
+        pageToUse = parsedPage;
+      }
+  
+      if (!totalResults) {
+        return <p className="paging">No results, try another search</p>;
+      }
+  
+      const lastPage = Math.ceil(totalResults / perPage);
+  
+      return (
+        <p className="paging">
+          <button
+            className="prev"
+            onClick={() => this.updatePage(-1)}
+            disabled={pageToUse === 1 ? "true" : ""}
```

Comment:
to fix the boolean issues in your console, change "true" to a boolean instead and this will get rid of your error!